From da0f0598d50d6f0a8eadb44202eb091ccb3c970c Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Fri, 29 Apr 2016 09:55:33 +0800
Subject: [PATCH 1194/5242] MLK-12928-11 mfd: pf1550: add irq ack for pf1550

commit  18ac895536235a866eda0a721e9c29cfaba67edf from
https://source.codeaurora.org/external/imx/linux-imx.git

add ".use_ack" ..etc for pf1550 irq, since we have to clear irq status in
pf1550, else no any more interrupt trigged.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/pf1550.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mfd/pf1550.c b/drivers/mfd/pf1550.c
index b60e822..e12d60d 100644
--- a/drivers/mfd/pf1550.c
+++ b/drivers/mfd/pf1550.c
@@ -86,8 +86,11 @@
 static const struct regmap_irq_chip pf1550_onkey_irq_chip = {
 	.name			= "pf1550-onkey",
 	.status_base		= PF1550_PMIC_REG_ONKEY_INT_STAT0,
+	.ack_base		= PF1550_PMIC_REG_ONKEY_INT_STAT0,
 	.mask_base		= PF1550_PMIC_REG_ONKEY_INT_MASK0,
 	.mask_invert		= false,
+	.use_ack                = 1,
+	.init_ack_masked	= 1,
 	.num_regs		= 1,
 	.irqs			= pf1550_onkey_irqs,
 	.num_irqs		= ARRAY_SIZE(pf1550_onkey_irqs),
@@ -105,8 +108,11 @@
 static const struct regmap_irq_chip pf1550_charger_irq_chip = {
 	.name			= "pf1550-charger",
 	.status_base		= PF1550_CHARG_REG_CHG_INT,
+	.ack_base		= PF1550_CHARG_REG_CHG_INT,
 	.mask_base		= PF1550_CHARG_REG_CHG_INT_MASK,
 	.mask_invert		= false,
+	.use_ack                = 1,
+	.init_ack_masked	= 1,
 	.num_regs		= 1,
 	.irqs			= pf1550_charger_irqs,
 	.num_irqs		= ARRAY_SIZE(pf1550_charger_irqs),
-- 
1.7.9.5

